casper/src/main/scala/coop/rchain/casper/util/comm/CasperPacketHandler.scala

```diff
@@ -378,6 +387,14 @@ object CasperPacketHandler extends CasperPacketHandlerInstances {
             }
       } yield none[Packet]
 
+    override def handleForkTipRequest(peer: PeerNode, ftr: ForkTipRequest): F[Option[Packet]] =
+      for {
+        _   <- Log[F].info(s"Received ForkTipRequest from $peer")
+        tip <- MultiParentCasper.forkTip
+        msg = Packet(transport.BlockMessage.id, tip.toByteString)
+        _   <- Log[F].info(s"Sending Block ${tip.blockHash} to $peer")
+      } yield Some(msg)
```

Comment:
You need to TL stream the block back instead of relying on the round trip mechanism that the comm team is trying to remove. @rabbitonweb feel free to correct if I'm mistaken.

Reply:
that is correct, `roundTrip` is problematic, usage is highly discouraged